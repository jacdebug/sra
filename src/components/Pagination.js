import React from 'react';
import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';
import { FlexContainer } from './Elements';

const Pagination = ({ paginationLinks }) =>
  <FlexContainer left>
    {paginationLinks.map(renderPaginationLink)}
  </FlexContainer>;

const renderPaginationLink = (link, index) =>
  <StyledLink
    disabled={!link.isEnabled}
    key={index}
    to={`/files${link.queryString}`}
  >
    {link.type}
  </StyledLink>;

export const StyledLink = styled(Link)`
  margin: 2px;
  padding: 3px 4px;
  text-decoration: none;
  border: 1px solid transparent;
  border-radius: 3px;
  border-color: rgba(175, 47, 47, 0.1);
  text-transform: uppercase;
  color: #444;
  &:first-child { margin-left: 0; }
  &:hover {  }
  &.selected { border-color: rgba(175, 47, 47, 0.2); }
  ${props =>
    props.disabled &&
    css`
      cursor: default;
      opacity: 0.5;
      pointer-events: none;
	`}
`;

export default Pagination;
