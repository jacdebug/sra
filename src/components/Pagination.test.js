import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from './Pagination';

const paginationLinksData = [
  {
    type: 'first',
    isEnabled: true,
    queryString: '?_page=1&name_like=',
    linkData: {
      _page: '1',
      name_like: '',
      rel: 'first',
      url: 'http://localhost:3003/files?_page=1&name_like=',
    },
  },
  { type: 'prev', isEnabled: false, queryString: '?', linkData: null },
  {
    type: 'next',
    isEnabled: true,
    queryString: '?_page=2&name_like=',
    linkData: {
      _page: '2',
      name_like: '',
      rel: 'next',
      url: 'http://localhost:3003/files?_page=2&name_like=',
    },
  },
  {
    type: 'last',
    isEnabled: true,
    queryString: '?_page=100&name_like=',
    linkData: {
      _page: '100',
      name_like: '',
      rel: 'last',
      url: 'http://localhost:3003/files?_page=100&name_like=',
    },
  },
];

it('renders without crashing', () => {
  shallow(<Pagination paginationLinks={paginationLinksData} />);
});

it('renders correctly', () => {
  const tree = shallow(<Pagination paginationLinks={paginationLinksData} />);
  expect(toJson(tree)).toMatchSnapshot();
});
