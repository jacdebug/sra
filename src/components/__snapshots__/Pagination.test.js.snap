// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div
  left={true}
>
  <Styled(Link)
    disabled={false}
    to="/files?_page=1&name_like="
  >
    first
  </Styled(Link)>
  <Styled(Link)
    disabled={true}
    to="/files?"
  >
    prev
  </Styled(Link)>
  <Styled(Link)
    disabled={false}
    to="/files?_page=2&name_like="
  >
    next
  </Styled(Link)>
  <Styled(Link)
    disabled={false}
    to="/files?_page=100&name_like="
  >
    last
  </Styled(Link)>
</styled.div>
`;
